.github/workflows/test.yml

```diff
@@ -30,7 +30,16 @@ jobs:
     - name: Build project
       run: npm run build
 
-    - name: Run tests
+    - name: Run server tests
+      run: npm run test:server
+
+    - name: Run component tests
+      run: npm run test:component
+
+    - name: Run site tests
+      run: npm run test:site
+
+    - name: Run all tests
```

Comment:
I'd just run `npm test`. It's redundant to run them individually and then all at once.